RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -129,13 +133,22 @@ const BeamSpotOnlineObjects* OnlineBeamSpotESProducer::checkSingleBS(const BeamS
   auto limitTime = std::chrono::microseconds((std::chrono::hours)timeThreshold_).count();
 
   // Check that the BS is within the timeThreshold, converges and passes the sigmaZthreshold
-  if (diffBStime1 < limitTime && bs1->sigmaZ() > sigmaZThreshold_ && bs1->beamType() == 2) {
+  if (diffBStime1 < limitTime && isGoodBS(bs1)) {
     return bs1;
   } else {
     return nullptr;
   }
 }
 
+// This method is used to check the quality of the beamspot fit
+bool OnlineBeamSpotESProducer::isGoodBS(const BeamSpotOnlineObjects* bs1) {
+  if (bs1->sigmaZ() > sigmaZThreshold_ && bs1->beamType() == 2 && bs1->beamWidthX() > sigmaXYThreshold_ &&
```

Comment:
since you have to change the PR anyway, how about using this `enum` (`BeamType::Tracker`) instead of the` 2`:

https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/BeamSpot/interface/BeamSpot.h#L24

Reply:
Personally I think the names in this `enum` are a bit misleading because 2 is the only "good" result, while the other codes all indicate a fit failure (for different reasons), but I never found the time to change it and check all the possible places where it should be updated.

So I think that using `BeamType::Tracker` here would not improve clarity in any case, and I propose this suggested change is postponed to a later PR together with an improvement of the enum description itself 😄

Reply:
2 is a magic number. I don't see reasons to postpone it. If you wish to change it, you can change this file as well everywhere after this is merged.

Reply:
(actually having the `enum` used here, will *help* you in identifying where things need to be changes, instead of scanning a `2` around (that would yield gazillions of false positives) so if anything this reinforces the request)

Reply:
Hi @mmusich ! Given that the search of each occurrence in the whole CMSSW of all the ``beamType() == 2`` and their replacement could be quite time consuming I would suggest to change it here and then I will create a separate pull request to replace all the others.

Reply:
> I would suggest to change it here and then I will create a separate pull request to replace all the others.

that's precisely what I was suggesting.

Reply:
Ok great! I misunderstood what you wrote then, sorry.

Reply:
so can you expect a commit shortly?